### ISSUE TITLE
Scheduler does not prevent duplicate show scheduling

### ISSUE THREAD

**Issue by CaptainHayashi** [UniversityRadioYork/MyRadio#49]
Duplicate show scheduling breaks the website (an issue against UniversityRadioYork/2013site is pending).  If MyRadio could check before committing schedule changes to database that they aren't going to overlap with existing scheduled shows, that would be very helpful as this is a very common issue.

Relevant IRC log:

```
16:00 <@DogeHayashi> sam_w: because the schedule renderer then tries to fill in URY Jukebox between the end of one show and the beginning of the other, finds that there's a negative gap (since the next show starts before the previous one finishes), and an assertion I wrote for that case is raised
16:01 <@DogeHayashi> sam_w: (because that should never happen, or at least it should never happen given decent input from the scheduler, but I didn't realise that could happen at the time I wrote the damned thing otherwise it wouldn't have just been an assertion)
16:02 <@DogeHayashi> sam_w: so, what needs to happen asap is the overlapping show needs sorting out, which is basically a SQL query of some form; long-term, the website wants to be able to treat negative gaps as a reason to display a friendly error message, and the scheduler wants to stop scheduling duplicate shows
16:03 <@DogeHayashi> the website end is probably searching for the assert for negative gaps and changing it to an exception, then catching the exception in the view code and dealing with it
16:03 <@DogeHayashi> the MyRadio end is f*** knows
```

**Comment by Flump5281** [comment]
Fixed in bootstrap.

**Comment by AJDurant** [comment]
Fixed by #280
